src/ChorizoServer.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using Chorizo.ProtocolHandler;
+
+namespace Chorizo
+{
+    public class Chorizo
+    {
+        public IServerStatus Status { get; set; }
+        public ISocketMachine SocketMachine;
+        public ICzoProtocolHandler ProtocolHandler;
+
+        private ServerConfig Config { get; }
+
+        public Chorizo()
+        {
+            ProtocolHandler = new CzoTelNetHandler();
+            SocketMachine = SocketMachine ?? new DotNetSocketMachine();
+            Config = new ServerConfig("HTTP", "localhost", 8000);
+        }
+
+        public Chorizo(int port, string protocol = null)
+        {
+            protocol = protocol ?? "HTTP";
```

Comment:
There seems to be some duping between these blocks—can they be simplified?

Reply:
Should be better now!